router/core/server.go

```diff
@@ -0,0 +1,684 @@
+		}
+	} else {
+		planCache = NewNoopExecutionPlanCache()
+	}
+
+	routerMetrics := NewRouterMetrics(&routerMetricsConfig{
+		metrics:             s.metricStore,
+		gqlMetricsExporter:  s.gqlMetricsExporter,
+		exportEnabled:       s.graphqlMetricsConfig.Enabled,
+		routerConfigVersion: routerConfigVersion,
+		logger:              s.logger,
+	})
+
+	var traceHandler *rtrace.Middleware
+	if s.traceConfig.Enabled {
+		spanStartOptions := []oteltrace.SpanStartOption{
+			oteltrace.WithAttributes(
+				otel.RouterServerAttribute,
+				otel.WgRouterRootSpan.Bool(true),
+			),
+		}
+
+		if s.traceConfig.WithNewRoot {
+			spanStartOptions = append(spanStartOptions, oteltrace.WithNewRoot())
+		}
+
+		traceHandler = rtrace.NewMiddleware(
+			func(r *http.Request) {
+				span := oteltrace.SpanFromContext(r.Context())
+				if attributes := baseAttributesFromContext(r.Context()); attributes != nil {
+					span.SetAttributes(attributes...)
+				}
+			},
+			otelhttp.WithSpanOptions(spanStartOptions...),
+			otelhttp.WithFilter(rtrace.CommonRequestFilter),
+			otelhttp.WithFilter(rtrace.PrefixRequestFilter(
+				[]string{s.healthCheckPath, s.readinessCheckPath, s.livenessCheckPath}),
+			),
+			// Disable built-in metricStore through NoopMeterProvider
+			otelhttp.WithMeterProvider(sdkmetric.NewMeterProvider()),
+			otelhttp.WithSpanNameFormatter(SpanNameFormatter),
+			otelhttp.WithTracerProvider(s.tracerProvider),
+		)
+	}
+
+	baseLogFields := []zapcore.Field{
+		zap.String("config_version", routerConfigVersion),
+	}
+
+	if featureFlagName != "" {
+		baseLogFields = append(baseLogFields, zap.String("feature_flag", featureFlagName))
+	}
+
+	// Request logger
+	requestLoggerOpts := []requestlogger.Option{
+		requestlogger.WithDefaultOptions(),
+		requestlogger.WithNoTimeField(),
+		requestlogger.WithFields(baseLogFields...),
+		requestlogger.WithRequestFields(func(request *http.Request) []zapcore.Field {
+			return []zapcore.Field{
+				zap.String("request_id", middleware.GetReqID(request.Context())),
+			}
+		}),
+	}
+
+	if s.ipAnonymization.Enabled {
+		requestLoggerOpts = append(requestLoggerOpts, requestlogger.WithAnonymization(&requestlogger.IPAnonymizationConfig{
+			Enabled: s.ipAnonymization.Enabled,
+			Method:  requestlogger.IPAnonymizationMethod(s.ipAnonymization.Method),
+		}))
+	}
+
+	var otelAttributesMappers []func(req *http.Request) []attribute.KeyValue
+
+	if s.traceConfig.SpanAttributesMapper != nil {
+		otelAttributesMappers = append(otelAttributesMappers, s.traceConfig.SpanAttributesMapper)
+	}
+
+	otelAttributesMappers = append(otelAttributesMappers, func(req *http.Request) []attribute.KeyValue {
+		return baseOtelAttributes
+	})
+
+	requestLogger := requestlogger.New(
+		s.logger,
+		requestLoggerOpts...,
+	)
+
+	// Enrich the request context with the subgraphs information and base OTEL attributes
+	httpRouter.Use(func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Should this be enabled by default or only in case of OTEL being enabled?

Reply:
Good point, I made it optionally.